UrlManager.php

```diff
@@ -197,7 +197,14 @@ public function createUrl($params)
             if (!$languageRequired && !$this->enableDefaultLanguageUrlCode && $language===$this->getDefaultLanguage()) {
                 return  $url;
             } else {
-                $url = rtrim($url, '/');
+                foreach ($this->rules as $rule) {
+                    if(in_array($rule->route, $params)) {
+                        if($rule->name=='' && $rule->suffix!='/') {
+                            $url = rtrim($url, '/');
+                        }
+                        break;
+                    }
+                }
```

Comment:
Yeah, I know it's not the best solution for that reason I added the break to stop once we found the currently url we are trying to create.